src/lib/ProjectM36/Base.hs

```diff
@@ -167,7 +212,18 @@ instance Hashable Relation where
       sortedAttrs = map snd (sortedAttributesIndices attrs)
       
 instance Binary Relation      
-  
+
+data DataFrame = DataFrame Attributes RelationTupleSet deriving (Show, Generic, Typeable)
```

Comment:
Maybe this should be `DataFrame Attributes [RelationTuple]` instead? It's not really a Set if it has ordering. The fact that `RelationTupleSet` merely wraps a list is merely an implementation/optimization detail.

Reply:
Sure. Thanks for pointing out. I will change at next commit.

Reply:
Thanks!

I don't know how to derive `Ord` for `Relation`s....
What is it like to order relation-valued attributes?
by TupleSet Number? That's one way to do it...
If I have a relation with one relation-valued attributes, then what the syntax would be?
`:showdataframe s orderby { ??? }`
relation-valued attribute must have an order first before `{a_rel_valued_attr_name DESC}`
So it may be `{ (attrName orderby {...}) DESC} ?

orderby seems doesn't affect relation property.
And limit and offset just change the tupleset number.
It seems dataframe and relation can be mutually recursive.

I don't have much experience in database so I don't think of any example about that.
Maybe in some case someone just want to interact a relation with the first ten tuple (like in a test or competition) so we need sorting.

Now I don't know how to implement the mutually-referencing datatype between relational expression and dataframe expression. If we need `back to relation`, then we need that, and some refactoring.